src/main/java/com/esri/samples/map/map_reference_scale/MapReferenceScaleController.java

```diff
@@ -0,0 +1,162 @@
+import com.esri.arcgisruntime.portal.Portal;
+import com.esri.arcgisruntime.portal.PortalItem;
+
+import javafx.fxml.FXML;
+import javafx.scene.control.CheckBox;
+import javafx.scene.control.ComboBox;
+import javafx.scene.control.Label;
+import javafx.scene.control.ProgressIndicator;
+import javafx.scene.layout.VBox;
+import javafx.util.StringConverter;
+
+public class MapReferenceScaleController {
+
+  @FXML
+  private MapView mapView;
+  @FXML
+  private Label scaleLabel;
+  @FXML
+  private Label loadingLabel;
+  @FXML
+  private ComboBox<Double> scaleComboBox;
+  @FXML
+  private VBox layerVBox;
+  @FXML
+  private VBox scaleVBox;
+  @FXML
+  private ProgressIndicator progressIndicator;
+
+  private ArcGISMap map;
+
+  @FXML
+  private void initialize() {
+
+    // access a web map as a portal item
+    Portal portal = new Portal("http://runtime.maps.arcgis.com");
+    PortalItem portalItem = new PortalItem(portal, "3953413f3bd34e53a42bf70f2937a408");
+
+    // create a map with the portal item
+    map = new ArcGISMap(portalItem);
+
+    // set the map to the map view
+    mapView.setMap(map);
+    map.setReferenceScale(250000);
+
+    scaleComboBox.setConverter(new StringConverter<>() {
+      @Override
+      public String toString(Double value) {
+        return "1:" + Math.round(value);
+      }
+
+      @Override
+      public Double fromString(String string) {
+        // not required
+        return null;
+      }
+    });
+
+    // create a label to display current scale of the map
+    mapView.addMapScaleChangedListener(event -> {
+      scaleLabel.setText("Current Map Scale 1:" + Math.round(mapView.getMapScale()));
+    });
+
+    map.addDoneLoadingListener(() -> {
+
+      progressIndicator.setVisible(false);
+
+      LayerList operationalLayers = map.getOperationalLayers();
+
+      for (Layer layer : operationalLayers) {
+        // create a checkbox per operational layer name
+        CheckBox checkBox = new CheckBox(layer.getName());
+        checkBox.setSelected(true);
+        // add the checkboxes to the VBox
+        layerVBox.getChildren().add(checkBox);
+
+        if (layer instanceof FeatureLayer) {
+
+          FeatureLayer featureLayer = (FeatureLayer) layer;
+          // set if the feature layer will honor the reference scale
+          checkBox.setOnAction(event -> {
+            featureLayer.setScaleSymbols(checkBox.isSelected());
+          });
+        }
+      }
+
+      scaleVBox.setVisible(true);
+      layerVBox.setVisible(true);
+
+      loadingLabel.setText("Apply Reference Scale");
+    });
+  }
+
+  /**
+   * Set the map reference scale when a selection is made using the combobox
+   */
+  @FXML
+  private void handleComboBoxSelection() {
+    setMapReferenceScale(scaleComboBox.getSelectionModel().getSelectedItem());
+  }
+
+  /**
+   * Takes the reference scale from the combobox, and sets it as the map's reference scale.
+   */
+  @FXML
+  private void handleScaleButtonClicked() {
```

Comment:
I was thinking it would be like this:

```java
   /**
   * Set the map's reference scale to the scale selected in the combo box.
   */
  @FXML
  private void handleComboBoxSelection() {
    
    map.setReferenceScale(scaleComboBox.getSelectionModel().getSelectedItem());
  }

  /**
   * Set the current viewpoint scale to the map's reference scale.
   */
  @FXML
  private void handleScaleButtonClicked() {
    
    // get the center of the current viewpoint extent
    Point centerPoint = mapView.getCurrentViewpoint(Viewpoint.Type.CENTER_AND_SCALE).getTargetGeometry().getExtent().getCenter();
    // get the map's current reference scale
    double currentReferenceScale = mapView.getMap().getReferenceScale();
    // set a viewpoint with the scale at the map's reference scale
    Viewpoint newViewPoint = new Viewpoint(centerPoint, currentReferenceScale);
    mapView.setViewpointAsync(newViewPoint);
  }
```

The first already sets the reference scale when a selection is made. No need to set it again in the second handler (just get it from the map).